File: src/main/java/uk/edu/populationfitness/models/genes/cache/LongIdentifier.java
package uk.edu.populationfitness.models.genes.cache;

import uk.edu.populationfitness.models.genes.GenesIdentifier;

/**
 * Identifies a bitset with a uniformly increasing long value.
 */
public class LongIdentifier implements GenesIdentifier {
    private static long globalIdentifier = 0;

    private final long value;

    public LongIdentifier(){
        value = ++globalIdentifier;
    }
    @Override
    public long asUniqueLong() {
        return value;
    }
}

File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/Cache/LongIdentifier.cs
namespace PopulationFitness.Models.Genes.Cache
{

    /**
     * Identifies a bitset with a uniformly increasing long value.
     */
    public class LongIdentifier : IGenesIdentifier
    {
        private static long _globalIdentifier = 0;

        private readonly long _value;

        public LongIdentifier()
        {
            _value = ++_globalIdentifier;
        }

        public long AsUniqueLong()
        {
            return _value;
        }
    }
}
